test/model/actions_test.dart

```diff
@@ -121,6 +145,45 @@ void main() {
       async.elapse(unregisterDelay - TestGlobalStore.removeAccountDuration);
       check(newConnection.isOpen).isFalse();
     }));
+
+    test('notifications are removed after logout', () => awaitFakeAsync((async) async {
+      await prepare();
+      testBinding.firebaseMessagingInitialToken = '123';
+      addTearDown(NotificationService.debugReset);
+      NotificationService.debugBackgroundIsolateIsLive = false;
+      await runWithHttpClient(NotificationService.instance.start);
+
+      check(testBinding.globalStore).accountIds.single.equals(eg.selfAccount.id);
+      const unregisterDelay = Duration(seconds: 5);
+      assert(unregisterDelay > TestGlobalStore.removeAccountDuration);
+      final newConnection = separateConnection()
+        ..prepare(delay: unregisterDelay, json: {'msg': '', 'result': 'success'});
```

Comment:
I think this logically goes after the "create a notification" steps. This step is about preparing the response that `logOutAccount` will get when it unregisters the token.